cmd/config-dir.go

```diff
@@ -117,6 +117,9 @@ func getCADir() string {
 }
 
 func createConfigDir() error {
+	if getConfigDir() == "" {
+		setConfigDir(mustGetDefaultConfigDir())
```

Comment:
This does not give any value i.e. its same as global `var configDir = &ConfigDir{dir: mustGetDefaultConfigDir()}`

If config directory is not passed in daemon mode, minio would crash here because of no user env available.

I don't see any solution other than having `/etc/minio` for daemon mode.

Below is the way it works
1. Get user home directory, if failed (in daemon mode mostly) use system wide `/etc/minio` as default.
2. User's home directory is higher priority over system wide `/etc/minio`

Reply:
The assumption was that in the case of daemon, we would require the user to specify a config directory. Let me see if I can get a directory for all OSes, in case the home directory is not available

Reply:
The issue this PR fixing is not finding home directory. All OSes have support for system level config directories

Reply:
There is no portable way to find home directory, that is why we use go-homedir. The problem is in macOS all techniques are disabled which are used by go-homedir so we get an error. 

User simply wanted us to do such that when config-dir is provided we are not still looking for home directory. Current master looks homedir regardless of config-dir is specified or not.. homedir outside the context of config-dir is not useful we don't use it anywhere else.

Reply:
If `config-dir` is not given in such system (and/or daemon mode), there are two option available.  Use system wide configuration directory or error out.

If error out option is chosen, this is not the right place to populate the global variable to find the error.

Reply:
We could default to a system wide default as well when all bets are off.